integration-cli/docker_api_images_test.go

```diff
@@ -121,3 +121,26 @@ func (s *DockerSuite) TestApiImagesDelete(c *check.C) {
 	c.Assert(status, check.Equals, http.StatusOK)
 	c.Assert(err, check.IsNil)
 }
+
+func (s *DockerSuite) TestApiImagesHistory(c *check.C) {
+	name := "test-api-images-history"
+	out, err := buildImage(name, "FROM hello-world\nENV FOO bar", false)
+	if err != nil {
+		c.Fatal(err)
+	}
+	defer deleteImages(name)
+	id := strings.TrimSpace(out)
+
+	status, body, err := sockRequest("GET", "/images/"+id+"/history", nil)
+	c.Assert(status, check.Equals, http.StatusOK)
+	c.Assert(err, check.IsNil)
+
+	var historydata []map[string]interface{}
```

Comment:
Please decode in this struct https://github.com/docker/docker/blob/master/api/types/types.go#L51